### PR TITLE
Prevent unnecessary initial updates / UI Refinements / Fixes

### PR DESCRIPTION
**Pull request by tsagadar**
Covers two more issues from the roadmap, additional UI improvements and bug fixes.

### CODE REVIEW

File: goosebit/api/firmware.py [hunk 1]
```diff
@@ -25,13 +25,27 @@ def search_filter(search_value):
...
+
+        result = {}
+        for entry in compat:
+            model = entry["model"]
+            revision = entry["revision"]
+            if model not in result:
+                result[model] = []
+            result[model].append(revision)
+
+        compatibility = "<br>".join([f"{model} - {', '.join(revisions)}" for model, revisions in result.items()])
```
Comment (b-rowan) [inline goosebit/api/firmware.py:38]: I think this might be better written as `dict[model, set[revision]]`.  This begs the question of restructuring the database to align with that format, namely having `Hardware` contain a `ManyToMany` field which would link to a separate model `Revision`.

File: goosebit/ui/templates/rollouts.html [hunk 2]
```diff
@@ -34,41 +34,67 @@ <h5 class="modal-title">Create Rollout</h5>
...
+                            Feed missing. Use "default" if working with a single
+                            feed.
+                        </div>
+                    </div>
+                    <div class="form-group mb-3">
+                        <label for="rollout-selected-flavor">Flavor</label>
+                        <input
+                            id="rollout-selected-flavor"
+                            class="form-control"
+                            placeholder="debug"
```
Comment (b-rowan) [inline goosebit/ui/templates/rollouts.html:65]: Placeholder seems wrong.

File: goosebit/ui/templates/rollouts.html [hunk 3]
```diff
@@ -34,41 +34,67 @@ <h5 class="modal-title">Create Rollout</h5>
...
+                            class="form-control"
+                            placeholder="Release 1"
+                        />
+                    </div>
+                    <div class="form-group mb-3">
+                        <label for="rollout-selected-feed">Feed</label>
+                        <input
+                            id="rollout-selected-feed"
+                            class="form-control"
+                            placeholder="qa"
```
Comment (b-rowan) [inline goosebit/ui/templates/rollouts.html:52]: Placeholder seems wrong.

File: goosebit/ui/templates/rollouts.html [hunk 4]
```diff
@@ -34,41 +34,67 @@ <h5 class="modal-title">Create Rollout</h5>
...
-                </button>
-            </div>
+            <form id="rollout-form" class="needs-validation" novalidate>
+                <div class="modal-body">
+                    <div class="form-group mb-3">
+                        <label for="rollout-selected-name">Name</label>
+                        <input
+                            id="rollout-selected-name"
+                            class="form-control"
+                            placeholder="Release 1"
```
Comment (b-rowan) [inline goosebit/ui/templates/rollouts.html:44]: Placeholder seems wrong.

File: pyproject.toml [hunk 5]
```diff
@@ -23,6 +23,7 @@ opentelemetry-distro = "^0.47b0"
 opentelemetry-instrumentation-fastapi = "^0.47b0"
 opentelemetry-exporter-prometheus = "^0.47b0"
 aiocache = "^0.12.2"
+httpx = "^0.27.0"
```
Comment (b-rowan) [inline pyproject.toml:26]: Why was this moved?  Not seeing it imported anywhere in the non-tests files.

**Review by b-rowan — CHANGES_REQUESTED** [review]

File: goosebit/ui/templates/rollouts.html (hunk 2 above)
Comment (tsagadar) [reply goosebit/ui/templates/rollouts.html:65]: "debug" would be a correct example of a flavor. But we decided to remove the "flavor" feature for an initial release -  so this will go away in a future PR.

File: goosebit/ui/templates/rollouts.html (hunk 3 above)
Comment (tsagadar) [reply goosebit/ui/templates/rollouts.html:52]: Again: a proper example for a feed. Idea for a better example - or should we not use any placeholder?

File: goosebit/ui/templates/rollouts.html (hunk 4 above)
Comment (tsagadar) [reply goosebit/ui/templates/rollouts.html:44]: Another example...

File: pyproject.toml (hunk 5 above)
Comment (tsagadar) [reply pyproject.toml:26]: It is used in `swdesc.py` (it was already here once - and I moved it back to test dependencies as I also did not see where it was used).

**Comment by b-rowan** [comment]
Placeholders make more sense now, I was thinking something more generic for them, but what you have should do fine for now.

File: goosebit/api/firmware.py (hunk 1 above)
Comment (tsagadar) [reply goosebit/api/firmware.py:38]: You mean returning the raw data vs. a pre-formatted string? Or simply that result[model] should not be of type list but rather set?

As for the db model: I don't see a big drawback in the current one - so hesitant to invest time there.

Comment (b-rowan) [reply goosebit/api/firmware.py:38]: I think just not a big fan of the preformatted string.  I would prefer to isolate the UI from the API as much as possible.

Comment (tsagadar) [reply goosebit/api/firmware.py]: Moved the formatting logic back to javascript (had it there initially)

**Review by b-rowan — APPROVED** [review]
:+1:
